tests/test_basics.py

```diff
@@ -49,9 +49,11 @@ def test_assert_ack_error(self):
                     raise salobj.AckError("no ack or error specified",
                                           ackcmd=salinfo.makeAckCmd(private_seqNum=3, ack=1, error=2))
 
-                with salobj.assertRaisesAckError(ack=1, error=2):
-                    raise salobj.AckError("matching ack and error",
-                                          ackcmd=salinfo.makeAckCmd(private_seqNum=4, ack=1, error=2))
+                result = "result for this exception"
+                with salobj.assertRaisesAckError(ack=1, error=2, result=result, result_contains=result[2:-2]):
+                    raise salobj.AckError("matching ack, error and result",
+                                          ackcmd=salinfo.makeAckCmd(private_seqNum=4, ack=1, error=2,
+                                                                    result=result))
```

Comment:
Do these `assertRaisesAckError` tests really need to be inside the async block? The `with assertRaises(Thing): raise Thing` feels tautological.

Reply:
I think it is crucial because this test is testing `assertRaisesAckError`. How else would you do it?

Reply:
I think maybe I just need a comment explaining that fact. "This is a test that assertRaisesAckError does X" would clarify it. That could also explain why it needs to be in the async block.

Reply:
I renamed the test to from `test_assert_ack_error` to `test_assert_raises_ack_error` and added this doc string: `Test the assertRaisesAckError function.`